src/main/sensors/barometer.c

```diff
@@ -493,10 +495,15 @@ float getBaroAltitude(void)
     return baro.altitude;
 }
 
+// Hack to avoid compiler optimising out the calibration code
+#if defined(USE_BARO_MS5611)
+__attribute__((optimize("-fno-finite-math-only")))
+#endif
 static void performBaroCalibrationCycle(const float altitude)
 {
-    static uint16_t cycleCount = 0;
-
+#if defined(USE_BARO_MS5611)
+    if (!isfinite(altitude)) return; // NaN or Inf is passed occasionally from MS5611 driver 
+#endif
```

Comment:
```suggestion
    if (!isfinite(altitude)) return; // NaN or Inf is passed occasionally from MS5611 driver 
```
With -ffinite-math-only, `isfinite()` is optimized out during compilation (all floats are assumed finite).